tofu/modules/harvester_host/locals.tf

```diff
@@ -0,0 +1,57 @@
+locals {
+  nonexistent_cloudinit_secrets     = [for cloudinit in var.cloudinit_secrets : cloudinit if length(cloudinit.user_data) > 0]
+  nonexistent_cloudinit_secrets_map = { for cloudinit in local.nonexistent_cloudinit_secrets : cloudinit.name => cloudinit }
+  existing_cloudinit_secrets        = [for cloudinit in var.cloudinit_secrets : cloudinit if !contains(local.nonexistent_cloudinit_secrets, cloudinit)]
+  existing_cloudinit_secrets_map    = { for cloudinit in local.existing_cloudinit_secrets : cloudinit.name => cloudinit }
+  default_init                      = <<EOT
+#cloud-config
+users:
+  - default
+  - ${var.user}
+password: ${var.password}
+disable_root: false
+chpasswd:
+  expire: false
+  users:
+    - {name: ${var.user}, password: ${var.password}, type: text}
+EOT
+  guest_agent_init                  = <<EOT
+package_update: true
+packages:
+  - qemu-guest-agent
+  - iptables
```

Comment:
Note both are already present with the NonCloud image, so you can remove all this part